ophyd/session/sessionmgr.py

```diff
@@ -228,7 +228,7 @@ def stop_all(self):
             self._run_engine.stop()
 
         for pos in self._registry['positioners'].values():
-            if pos.moving is True:
+            if pos.connected and pos.moving is True:
```

Comment:
`if pos.connected and pos.moving is True:` is equivalent to `if pos.connected and pos.moving`, perhaps use the latter